include/modem/location.h

```diff
@@ -0,0 +1,341 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+/**
+ * @file location.h
+ * @brief Public APIs for the Location library.
+ * @defgroup location Location library
+ * @{
+ */
+
+#ifndef LOCATION_H_
+#define LOCATION_H_
+
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+#if defined(CONFIG_LOCATION_METHOD_GNSS_AGPS_EXTERNAL)
+#include <nrf_modem_gnss.h>
+#endif
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/** Maximum number of location methods that the library can handle in the location request. */
+#define LOCATION_MAX_METHODS 3
+
+/** Location method. */
+enum location_method {
+	/** LTE cellular positioning. */
+	LOCATION_METHOD_CELLULAR = 1,
+	/** Global Navigation Satellite System (GNSS). */
+	LOCATION_METHOD_GNSS,
+	/** Wi-Fi positioning. */
+	LOCATION_METHOD_WIFI,
+};
+
+/** Event IDs. */
+enum location_event_id {
+	/** Location update. */
+	LOCATION_EVT_LOCATION,
+	/** Getting location timed out. */
+	LOCATION_EVT_TIMEOUT,
+	/** An error occurred when trying to get the location. */
+	LOCATION_EVT_ERROR,
+	/**
+	 * GNSS is requesting assistance data. Application should obtain the data and send it to
+	 * location_agps_data_process().
+	 */
+	LOCATION_EVT_GNSS_ASSISTANCE_REQUEST
+};
+
+/** Location accuracy. */
+enum location_accuracy {
+	/** Allow lower accuracy to conserve power. */
+	LOCATION_ACCURACY_LOW,
+	/** Normal accuracy. */
+	LOCATION_ACCURACY_NORMAL,
+	/** Allow higher power consumption to improve accuracy. */
+	LOCATION_ACCURACY_HIGH
```

Comment:
We still need to discuss what to do with HIGH_ACCURACY.

Reply:
@hiltunent will remove high accuracy and we'll investigate and test later if this implementation (as it is or modified) will really improve accuracy. Right now we don't have evidence that it would be better than `LOCATION_ACCURACY_NORMAL`.

Reply:
LOCATION_ACCURACY_HIGH has been removed.
High accuracy example in the sample has been changed to low accuracy.